### ISSUE TITLE
Compatibility issue with eslint-plugin-vuejs-accessibility

### ISSUE THREAD

**Issue by constgen** [ota-meshi/eslint-plugin-jsonc#31]
When I try to use https://github.com/vue-a11y/eslint-plugin-vuejs-accessibility in the config together with `eslint-plugin-jsonc` something doesn't work as expected. `eslint-plugin-vuejs-accessibility` utilizes `vue-eslint-parser` (its peer dependency). Thats why when JSON files are linted it fails with an exception

```
Oops! Something went wrong! :(

ESLint: 7.18.0

TypeError: Error while loading rule 'vuejs-accessibility/interactive-supports-focus': context.parserServices.defineTemplateBodyVisitor is not a function
```
The ESLint config is like this (simplified example):

```json5
{
  "plugins": [
    "jsonc",
    "vue",
    "vuejs-accessibility"
  ],
  "parser": "vue-eslint-parser",
  "overrides": [
    {
      "files": ["*.json", "*.jsonc", "*.json5", "*.json6"],
      "parser":"jsonc-eslint-parser",
      "rules": {
        "jsonc/auto": "error"
      }
    }
  ],
  "rules": {
     "jsonc/indent": "error",
     "vuejs-accessibility/accessible-emoji": "error",
     // many rules
  }
}
```

And of course ESLint fails as we have overridden the parser for JSON files

I tried also another approach as Vue Plugin suggest - use `"parserOptions.parser"`

```json5
{
  "plugins": [
    "jsonc",
    "vue",
    "vuejs-accessibility"
  ],
  "parser": "vue-eslint-parser",
  "overrides": [
    {
      "files": ["*.json", "*.jsonc", "*.json5", "*.json6"],
      "parserOprions": {
        "parser":"jsonc-eslint-parser",
      },
      "rules": {
        "jsonc/auto": "error"
      }
    }
  ],
  "rules": {
     "jsonc/indent": "error",
     "vuejs-accessibility/accessible-emoji": "error",
     // many rules
  }
}
```
Now plugins work. Many core rules are also linted. But `jsonc` rules are not checked any more. For example `"jsonc/indent"` errors are not reported.

### Expected
One of both
1. `vuejs-accessibility/*` rules should not throw when the parser is not `vue-eslint-parser` and just ignore
2.  All `jsonc/*` rules are applied when configured with `{ "parser": "vue-eslint-parser", "parserOprions":  { "parser":"jsonc-eslint-parser" } }`

**Comment by ota-meshi** [comment]
Thank you for reporting this issue.
I think it's a bug in eslint-plugin-vuejs-accessibility that catches the error, but this plugin also had a bug that didn't work well when mixed parsers. I will fix this bug.

**Comment by constgen** [comment]
Absolutely agree with your answer. Have tested it and it works after your fix. Thanks for a quick feedback.

I decided to test integratino with other plugins I always use. And again some rules work and some not.

### This works

`unicorn/filename-case` reports an error as expected

```
 1:1   error    Filename is not in kebab case or pascal case. Rename it to `json-config copy.json`, `json-config Copy.json`, `JsonConfig copy.json`, or `JsonConfig Copy.json` 
```

### This doesn't work

I expect this to report but it doesn't `sonarjs/no-duplicate-string`. 

I think I remember it worked **before** `v0.8.0`. This particular case is not urgent for me as I would like to disable this rule for JSON any way. But it is good to know for you about some minor issues

**Comment by ota-meshi** [comment]
It is intentional behavior in this plugin that most rules using AST in the ESLint core and other plugins do not work in JSON.
This means that when used with many plugins, there are few false positives in JSON.
However, if you want to apply the rules of the ESLint core and other plugins to JSON, we need to create a new rule.
If you have a rule you want, open an issue in this repository and suggest it.
